src/librustc/ich/impls_hir.rs

```diff
@@ -391,13 +391,14 @@ impl<'a> HashStable<StableHashingContext<'a>> for hir::TraitCandidate {
                                           hcx: &mut StableHashingContext<'a>,
                                           hasher: &mut StableHasher<W>) {
         hcx.with_node_id_hashing_mode(NodeIdHashingMode::HashDefPath, |hcx| {
-            let hir::TraitCandidate {
+            let &hir::TraitCandidate {
```

Comment:
```suggestion
            let hir::TraitCandidate {
```
Nit: IIUC, binding autoref should work here.

Reply:
Ah, it did. Fixed and pushed.